### PR TITLE
Add mattresses

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
 ```SUMMARY: content "Adds mattresses"```

#### Purpose of change
currently beds are made with 2x4 and blankets, which doesn't seem very comfortable, and makes the makeshift bed an almost totally irrelevant option.

#### Describe the solution
Adds mattresses, heavy bulky items needed to make a proper bed.

Beds can now be built in two stages - bed frame and mattress. The bed frame is the tricky part, adding or removing a mattress is quite easy (I left the option to do it all at once if you have everything on hand).  If you don't have a bed frame, you can just deploy a mattress on the floor and sleep there. It's not quite as cozy but it's better than a makeshift bed.

I will look into placing mattresses into some common spawn locations, but really there should be no shortage of them simply by deconstructing existing beds.

- [x] add mattresses
- [x] change bed recipe and deconstruct
- [x] make an uncraft recipe for mattresses that yields rags and sheets
- [x] make floor mattress deployable
- [x] make bed build in two stages if desired, and make it possible to just strip the mattress from a bed frame.

#### Describe alternatives you've considered
This is all a little muddied by the fact that really beds should be a multi tile furniture, and one mattress and one construction should make a 2x1 bed (that can then be dragged around and doesn't pull apart). I considered having a mattress be an item you activate to get two mattress sections, each making half of a single bed, but it felt way too artificial. Since the player can currently sleep on a 1x1 bed, I'm going to leave it like this and players can just get a lot of mattresses from pre-existing beds. The limiting factor with mattresses is going to be transporting them, not finding them anyway.

#### Additional context
Another piece of flat pack furniture, which in turn is part of finishing the beggar quests. It's so strange how this stuff branches out.

### CODE REVIEW

**Comment by Rail-Runner** [comment]
Possibly add a crafting recipe for mattresses too, for those who would want to build a quality bed?

**Comment by I-am-Erk** [comment]
I'm skeptical that a survivor could build a commercial quality mattress from scratch. I think any mattress crafting recipe would basically result in a makeshift bed.

**Comment by Rail-Runner** [comment]
> I'm skeptical that a survivor could build a commercial quality mattress from scratch. I think any mattress crafting recipe would basically result in a makeshift bed.

Could be feather-filled one. I heard such mattresses were used for very expensive beds in medieval times, so should be good enough.

**Comment by Amoebka** [comment]
From what I understand mattresses currently can only be used to make normal beds, which requires fab, tailoring and tools. I believe they should be deployable by themselves (like braziers) because, realistically speaking, just dropping a mattress on the floor makes for a perfectly serviceable bed.

**Comment by paulenka-aleh** [comment]
Could you add an option to place them to sleep on, rollmat-like? Good mattress or the floor is more comfortable option than a vehicle seat or even a makeshift bed. Even though it's not very practical in-game it's definetly an option to use before constructing real bed.

Btw, I would later on place them at some survivors camps or places. E.g. if I was hiding in a pantry-like basement I might drag mattress below before night-time. Or a refugee center which might be short of beds and short of tools or hands to build them while fortifications is a priority can have some. But that sounds like another story?

**Comment by I-am-Erk** [comment]
I can make them deployable, might get around to that in this PR.

**Comment by Amoebka** [comment]
Another thing to maybe consider is making the bed deconstruction process consist of 2 steps. First is removing the mattress, which normally shouldn't require any tools or time (I've never seen a bed that required me to use a hammer/screwdriver to remove the mattress for cleaning), and should produce a matress and leave behind an empty bed frame. The frame itself can further be deconstructed into wood and nail using tools and some time.

**Comment by I-am-Erk** [comment]
It is a bit weird that beds require tools to remove the mattress, but I'm not aware of any way to make furniture deconstruction leave a new type of furniture behind. I toyed with making a construction pattern go bed frame -> bed, but reversing that doesn't appear to work.

Made your deployable mattresses though. They're in between regular and makeshift beds.

**Comment by Amoebka** [comment]
It could probably be done by defining a new "examine_action" for bed furniture and creating a new member function for iexamine. Having an examine action that yields an item and changes the examined furniture into a different type is potentially useful for a lot more than just beds and mattresses, but it also sounds like something completely outside of the scope for this pr.

**Comment by I-am-Erk** [comment]
I'm definitely not going to add a new examine action.

However I found the json tag I need incidentally while adding cardboard stuff to another PR, so I'll probably go back to the original bed frame plan.

**Comment by I-am-Erk** [comment]
There you go, you can easily strip the mattress off a bed if you like.

File: data/json/construction.json [hunk 1]
```diff
@@ -1510,6 +1510,27 @@
...
+    "pre_special": "check_empty",
+    "post_terrain": "f_bed_frame"
+  },
+  {
+    "type": "construction",
+    "description": "Add Mattress to Bed Frame",
+    "category": "FURN",
+    "required_skills": [ [ "fabrication", 0 ] ],
+    "time": 5,
+    "components": [ [ "mattress: ] ],
```
Comment (anothersimulacrum) [inline data/json/construction.json]: This line looks like it's missing a few things

Comment (I-am-Erk) [reply data/json/construction.json]: Is that from an older merge?

Comment (anothersimulacrum) [reply data/json/construction.json]: Looks so, my bad.

**Comment by kevingranade** [comment]
You do not seem to have uploaded the f_floor_mattress definition.

**Comment by I-am-Erk** [comment]
Huh.

I bet I put it in the wrong PR.

**Pull request merged**